fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -232,10 +279,22 @@ private void deleteDocument(Tuple groupingKey, byte[] primaryKey) throws IOExcep
             throw new RecordCoreException("Issue deleting old index keys", "oldRecord", oldRecord, e);
         }
 
+        Map<Tuple, AnalyzingInfixSuggester> suggesterMap = new HashMap<>();
+
         // update new
         try {
             for (Map.Entry<Tuple, List<LuceneDocumentFromRecord.DocumentField>> entry : newRecordFields.entrySet()) {
-                writeDocument(entry.getValue(), entry.getKey(), newRecord.getPrimaryKey().pack());
+                if (autoCompleteEnabled) {
+                    final AnalyzingInfixSuggester suggester = getSuggester(entry.getKey());
+                    suggesterMap.put(entry.getKey(), suggester);
```

Comment:
can you put this line in the getSuggester method and make reduce this to an inline if for the argument? I'm not sure if thats easier but it would put these all on one line. something like 
``` writeDocument(entry.getValue(), entry.getKey(), newRecord.getPrimaryKey().pack(), autoCompleteEnabled ? getSuggester(entry.getKey()) : null); 
```

Reply:
Yeah put the initialization of suggester into the ```writeDocument``` now.